cpp/src/cylon/table.cpp

```diff
@@ -915,19 +922,22 @@ Status DoDistributedSetOperation(std::shared_ptr<cylon::CylonContext> &ctx,
   }
 }
 
-Status Table::DistributedUnion(std::shared_ptr<Table> &left, std::shared_ptr<Table> &right,
+Status DistributedUnion(std::shared_ptr<Table> &left, std::shared_ptr<Table> &right,
 							   std::shared_ptr<Table> &out) {
-  return DoDistributedSetOperation(left->ctx, &Table::Union, left, right, out);
+  auto ctx = left->GetContext();
+  return DoDistributedSetOperation(ctx, &Union, left, right, out);
```

Comment:
Lets remove the ctx from the DoDistributedSetOperation method signature. I think its redundant here.